src/api/nlopt-in.hpp

```diff
@@ -89,10 +98,29 @@ namespace nlopt {
     typedef struct {
       opt *o;
       mfunc mf; func f; void *f_data;
+      functor_type functor;
       vfunc vf;
       nlopt_munge munge_destroy, munge_copy; // non-NULL for SWIG wrappers
     } myfunc_data;
 
+    void* alloc_myfunc_data_with_nulls() {
+      // need to return void* otherwise SWIG doesn't compile because
+      // myfunc_data is private
+      myfunc_data *d = new myfunc_data;
+      if (!d) throw std::bad_alloc();
+
+      d->o             = this;
+      d->f             = NULL;
+      d->f_data        = NULL;
+      d->mf            = NULL;
+      d->vf            = NULL;
+      d->munge_destroy = NULL;
+      d->munge_copy    = NULL;
```

Comment:
Can't you just use `new myfunc_data()` (note [parentheses](https://stackoverflow.com/questions/620137/do-the-parentheses-after-the-type-name-make-a-difference-with-new)) to [automatically zero-initialize](https://en.cppreference.com/w/cpp/language/zero_initialization)?